### PR TITLE
Cpicks

### PR DESCRIPTION
**Pull request by nityeshaga**
`cherry-pick`ed commits from #4263 and #4271 and made the pending changes

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by prachi1210** [comment]
@nityeshaga you should complete the checklist in the PR description which basically indicates that you ran coala over your code locally. Also, please rebase your branch.

**Comment by nityeshaga** [comment]
Oh yes! Sorry I totally missed that! :-p

**Comment by nityeshaga** [comment]
All checks have passed now! This should be ready for a merge unless a maintainer can acknowledge that I should `cherry-pick` other PRs that have been inactive for over a month to this branch. @jayvdb @sils

File: docs/Developers/Newcomers_Guide.rst [hunk 1]
```diff
@@ -126,22 +126,14 @@ Now you are ready to join coala community at `coala gitter <https://coala.io/cha
...
-----------------------------------------------
-
-Let us know on gitter that you are interested in contributing and ask for an
-invitation to our org. This is the first step towards contributing.
-A maintainer will command ``cobot`` (our gitter bot) to invite
-you to the Newcomer team.
-The invitation will be sent by email and you will have to accept
-it to join. If you don't see the invitation, accept it on our `team page
-<https://github.com/coala>`__.
+When you say "Hello World" in chat ``cobot`` (our gitter bot) will invite you
```
Comment (sils) [inline docs/Developers/Newcomers_Guide.rst]: cobot -> corobo

Comment (nityeshaga) [inline docs/Developers/Newcomers_Guide.rst]: do i create a new commit for changing all mentions of cobot to corobo or include it in one of the already existing ones?

**Comment by sils** [comment]
reack https://github.com/coala/coala/pull/4437/commits/ccebcfa4e457cee27d52d88e0377d3afde8ef15c https://github.com/coala/coala/pull/4437/commits/b9a14601e07ed354fad761c4e3877e7fa6c70c5a https://github.com/coala/coala/pull/4437/commits/75755f0e65d08d4ec28e0b2ef2d585c8c310c6be

**Comment by sils** [comment]
@nityeshaga needs a rebase though

**Comment by nityeshaga** [comment]
rebase done!

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/4437/commits/317cbe63b2eb818453e214f4c2c87d58c9353e2b https://github.com/coala/coala/pull/4437/commits/d8f8e42f42cd1645ff1cd2f36e97ff783ff1d227 https://github.com/coala/coala/pull/4437/commits/538d6820caec005dfb47546e6cbaa19ffbaa2a66

**Comment by sils** [comment]
(reack)

**Comment by sils** [comment]
@rultor merge

**Pull request merged**